Support building from tar.gz

## Agent Version and Platform
Agent version: 2.136.1, using the tar.gz package from https://github.com/Microsoft/vsts-agent/releases/tag/v2.136.1

Machine: Linux amd (Raspberry Pi 3+) running Raspbian.  Linux 4.14.52-v7+ armv7l

## What's not working?
Builds fail; they look for information (the version information?) in the git repository.  Since this source tree was expanded from a tarball instead of a git repository, this fails.

```
% ./dev.sh b
~/tmp/vsts-agent-2.136.1/src ~/tmp/vsts-agent-2.136.1/src
Prepend /home/ethomson/tmp/vsts-agent-2.136.1/src/../_dotnetsdk/2.1.300 to %PATH%


-----------------------------------------
Dotnet SDK Version
-----------------------------------------
2.1.300


-----------------------------------------
Pre-cache external resources for linux platform ...
-----------------------------------------
Download exists: TEE-CLC-14.134.0.zip
Download exists: vso-task-lib.tar.gz
Download exists: node-v6.10.3-linux-x64.tar.gz


-----------------------------------------
Building ...
-----------------------------------------
Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  fatal: Not a git repository (or any of the parent directories): .git
/home/ethomson/tmp/vsts-agent-2.136.1/src/dir.proj(4,5): error MSB3073: The command "git rev-parse HEAD" exited with code 128.
Failed: build
~/tmp/vsts-agent-2.136.1/src
```

The obvious workaround is to clone the git repository and check out v2.136.1, but it's awkward that one can't simply build from the source release.

--- a/src/Test/L0/ConstantGenerationL0.cs
+++ b/src/Test/L0/ConstantGenerationL0.cs
@@ -26,5 +26,15 @@ namespace Microsoft.VisualStudio.Services.Agent.Tests
             Assert.True(BuildConstants.Source.CommitHash.Length == 40, $"CommitHash should be SHA-1 hash {BuildConstants.Source.CommitHash}");
             Assert.True(validPackageNames.Contains(BuildConstants.AgentPackage.PackageName), $"PackageName should be one of the following '{string.Join(", ", validPackageNames)}', current PackageName is '{BuildConstants.AgentPackage.PackageName}'");
         }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Agent")]
+        public void BuiltFromGitNotFromTarball()
+        {
+            // for local runs, it's OK for this test to fail
+            // but we would not want to ship an agent with an empty commit ID
+            Assert.True(BuildConstants.Source.CommitHash != new string('0', 40), $"CommitHash should be non-empty");
+        }
     }
 }

--- a/src/Test/L0/ConstantGenerationL0.cs
+++ b/src/Test/L0/ConstantGenerationL0.cs
@@ -30,11 +30,12 @@ namespace Microsoft.VisualStudio.Services.Agent.Tests
         [Fact]
         [Trait("Level", "L0")]
         [Trait("Category", "Agent")]
-        public void BuiltFromGitNotFromTarball()
+        public void ReleaseBuiltFromGitNotFromTarball()
         {
-            // for local runs, it's OK for this test to fail
-            // but we would not want to ship an agent with an empty commit ID
+#if !DEBUG
+            // don't ship an agent with an empty commit ID
             Assert.True(BuildConstants.Source.CommitHash != new string('0', 40), $"CommitHash should be non-empty");
+#endif
         }
     }
 }

--- a/src/dir.proj
+++ b/src/dir.proj
@@ -14,10 +14,11 @@
   </PropertyGroup>
 
   <Target Name="GenerateConstant">
-    <Exec Command="git rev-parse HEAD" ConsoleToMSBuild="true">
+    <Exec Command="git rev-parse HEAD" ConsoleToMSBuild="true" IgnoreExitCode="true">
       <Output TaskParameter="ConsoleOutput" PropertyName="GitInfoCommitHash" />
     </Exec>
-    <Message Text="Building $(GitInfoCommitHash) --- $(PackageRuntime)" Importance="high"/>
+    <Message Text="Building $(GitInfoCommitHash) --- $(PackageRuntime)" Importance="high" Condition=" !$(GitInfoCommitHash.Contains('fatal')) "/>
+    <Message Text="Building from non-Git sources --- $(PackageRuntime)" Importance="high" Condition=" $(GitInfoCommitHash.Contains('fatal')) "/>
 
     <ItemGroup>
         <BuildConstants Include="namespace Microsoft.VisualStudio.Services.Agent"/>
@@ -26,7 +27,8 @@
         <BuildConstants Include="%20%20%20%20{"/>
         <BuildConstants Include="%20%20%20%20%20%20%20%20public static class Source"/>
         <BuildConstants Include="%20%20%20%20%20%20%20%20{"/>
-        <BuildConstants Include="%20%20%20%20%20%20%20%20%20%20%20%20public static readonly string CommitHash = %22$(GitInfoCommitHash)%22%3B"/>
+        <BuildConstants Include="%20%20%20%20%20%20%20%20%20%20%20%20public static readonly string CommitHash = %22$(GitInfoCommitHash)%22%3B" Condition=" !$(GitInfoCommitHash.Contains('fatal')) "/>
+        <BuildConstants Include="%20%20%20%20%20%20%20%20%20%20%20%20public static readonly string CommitHash = %220000000000000000000000000000000000000000%22%3B" Condition=" $(GitInfoCommitHash.Contains('fatal')) "/>
         <BuildConstants Include="%20%20%20%20%20%20%20%20}%0A"/>
         <BuildConstants Include="%20%20%20%20%20%20%20%20public static class AgentPackage"/>
         <BuildConstants Include="%20%20%20%20%20%20%20%20{"/>

